internal/server/iigointernal/base_judge.go

```diff
@@ -0,0 +1,271 @@
+package iigointernal
+
+import (
+	"math/rand"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/pkg/errors"
+)
+
+// base Judge object
+type BaseJudge struct {
+	id                int
+	budget            int
+	presidentSalary   int
+	BallotID          int
+	ResAllocID        int
+	speakerID         int
+	presidentID       int
+	evaluationResults map[int]roles.EvaluationReturn
+	clientJudge       roles.Judge
+}
+
+func (j *BaseJudge) init() {
+	j.BallotID = 0
+	j.ResAllocID = 0
+}
+
+// Withdraw president salary from the common pool
+// Call common withdraw function with president as parameter
+func (j *BaseJudge) withdrawPresidentSalary(gameState *gamestate.GameState) error {
+	var presidentSalary = int(rules.VariableMap["presidentSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(presidentSalary, gameState)
+	if withdrawError != nil {
+		featureJudge.presidentSalary = presidentSalary
```

Comment:
took me a whole min to try to find where `featureJudge` is... maybe let's not mutate package-level globals? I'm definitely not a fan of those! makes code very hard to debug.

Reply:
I second this